File: frontend/lib/models/resume.dart
/// Metadata for an uploaded resume PDF.
class Resume {
  final String id;
  final String name;
  final String fileName;
  final String uploadDate; // ISO date string e.g. "2026-06-10"
  final String? notes;

  const Resume({
    required this.id,
    required this.name,
    required this.fileName,
    required this.uploadDate,
    this.notes,
  });

  factory Resume.fromJson(Map<String, dynamic> json) {
    return Resume(
      id: json['id'] as String? ?? '',
      name: json['name'] as String? ?? '',
      fileName: json['fileName'] as String? ?? '',
      uploadDate: json['uploadDate'] as String? ?? '',
      notes: json['notes'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'fileName': fileName,
        'uploadDate': uploadDate,
        'notes': notes,
      };

  Resume copyWith({
    String? id,
    String? name,
    String? fileName,
    String? uploadDate,
    String? notes,
  }) {
    return Resume(
      id: id ?? this.id,
      name: name ?? this.name,
      fileName: fileName ?? this.fileName,
      uploadDate: uploadDate ?? this.uploadDate,
      notes: notes ?? this.notes,
    );
  }
}

File: backend/demo/src/main/java/com/jobtracker/demo/model/Resume.java
package com.jobtracker.demo.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * Metadata for an uploaded resume PDF.
 * The actual file is stored on the local filesystem under /uploads/resumes/.
 */
@Document(collection = "resumes")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Resume {

    @Id
    private String id;

    /** User-facing label, e.g. "SDE Resume v3" */
    private String name;

    /** Stored filename on disk, e.g. "uuid-sde-resume-v3.pdf" */
    private String fileName;

    /** ISO-8601 date string of when the file was uploaded */
    private String uploadDate;

    /** Optional notes about this resume version */
    private String notes;
}
